cypress/support/api_commands/channels.js

```diff
@@ -0,0 +1,148 @@
+function reqChannelBySlot(slot) {
+  return cy.request({
+    method: 'GET',
+    url: `/channels/slot:${slot}`,
+    headers: {
+      accept: 'application/json',
+    }
+  }).its('body');
+}
+Cypress.Commands.add('reqChannelBySlot', reqChannelBySlot);
+
+function reqSetChannelOutput(slot, value) {
+  return cy.request({
+    method: 'POST',
+    url: '/channels/slot:' + slot + '/output',
+    body: { value },
+    headers: {
+      'content-type': 'application/json',
+      accept: 'application/json',
+    }
+  });
+}
+Cypress.Commands.add('reqSetChannelOutput', reqSetChannelOutput);
+
+function reqSetChannelLiveID(slot, value) {
+  return cy.request({
+    method: 'POST',
+    url: '/channels/slot:' + slot + '/liveId',
+    body: { value },
+    headers: {
+      'content-type': 'application/json',
+      accept: 'application/json',
+    }
+  });
+}
+Cypress.Commands.add('reqSetChannelLiveID', reqSetChannelLiveID);
+
+function reqChannels() {
+  return cy.request({
+    method: 'GET',
+    url: '/channels',
+    headers: {
+      accept: 'application/json',
+    }
+  }).as('channels');
+}
+Cypress.Commands.add('reqChannels', reqChannels);
+
+function reqChannelById(id) {
+  return cy.request({
+    method: 'GET',
+    url: '/channels/' + id,
+    headers: {
+      accept: 'application/json',
+    }
+  }).as('channel');
+}
+Cypress.Commands.add('reqChannelById', reqChannelById);
+
+// Only able to update name?
+function updateChannel(id, name, opts = {}) {
+  return cy.request({
+    method: 'PUT',
+    url: '/channels/' + id,
+    body: {
+      name,
+    },
+    headers: {
+      'content-type': 'application/json',
+      accept: 'application/json',
+    },
+    ...opts
+  }).as('updatedChannel');
+}
+Cypress.Commands.add('updateChannel', updateChannel);
+
+function reqChannelLatestReading(id) {
+  return cy.request({
+    method: 'GET',
+    url: '/channels/' + id + '/latest_reading',
+    headers: {
+      accept: 'application/json',
+    }
+  }).as('reading');
+}
+Cypress.Commands.add('reqChannelLatestReading', reqChannelLatestReading);
+
+function reqAllChannelReadingsByRange(start, end) {
+  return cy.request({
+    method: 'GET',
+    url: '/channels/readings?start=' + start + '&end=' + end,
+    headers: {
+      accept: 'application/json',
+    }
+  }).as('readings');
+}
+Cypress.Commands.add('reqAllChannelReadingsByRange', reqAllChannelReadingsByRange);
+
+function reqAllChannelReadingsByLatest(latest) {
+  return cy.request({
+    method: 'GET',
+    url: '/channels/readings?latest=' + latest,
+    headers: {
+      accept: 'application/json',
+    }
+  }).as('readings');
+}
+Cypress.Commands.add('reqAllChannelReadingsByLatest', reqAllChannelReadingsByLatest);
+
+function reqChannelReadingsByRange(id, start, end) {
+  return cy.request({
+    method: 'GET',
+    url: '/channels/' + id + '/readings?start=' + start + '&end=' + end,
```

Comment:
while it isn't super necessary here, cypress `request()` includes `qs` option which will properly escape query params.  https://docs.cypress.io/api/commands/request#Syntax

Example:
```suggestion
    url: `/channels/${id}/readings`,
    qs: {start, end},
```

Reply:
Or for that matter you could just use `...opts` and the caller would provide the option i.e.

```
function reqChannelReadings(id, opts) { ... }
```
usage
```
cy.requestChannelReadings(1, {qs: {start, end}});
```